libcaf_test/caf/test/dsl.hpp

```diff
@@ -689,16 +718,361 @@ class test_coordinator_fixture {
+          break;
+      }
+      return true;
+    }
+
+    /// Tries to execute a single event in LIFO order.
+    bool try_run_once_lifo() {
+      if (jobs.empty())
+        return false;
+      if (jobs.size() >= 2)
+        std::rotate(jobs.rbegin(), jobs.rbegin() + 1, jobs.rend());
+      return try_run_once();
+    }
+
+    /// Executes a single event in FIFO order or fails if no event is available.
+    void run_once() override {
+      if (jobs.empty())
+        CAF_RAISE_ERROR("No job to run available.");
+      try_run_once();
+    }
+
+    /// Executes a single event in LIFO order or fails if no event is available.
+    void run_once_lifo() {
+      if (jobs.empty())
+        CAF_RAISE_ERROR("No job to run available.");
+      try_run_once_lifo();
+    }
+
+    /// Executes events until the job queue is empty and no pending timeouts are
+    /// left. Returns the number of processed events.
+    size_t run(size_t max_count = std::numeric_limits<size_t>::max()) {
+      size_t res = 0;
+      while (res < max_count && try_run_once())
+        ++res;
+      return res;
+    }
+
+    /// Returns whether at least one pending timeout exists.
+    bool has_pending_timeout() const {
+      return clock_.has_pending_timeout();
+    }
+
+    /// Tries to trigger a single timeout.
+    bool trigger_timeout() {
+      return clock_.trigger_timeout();
+    }
+
+    /// Triggers all pending timeouts.
+    size_t trigger_timeouts() {
+      return clock_.trigger_timeouts();
+    }
+
+    /// Advances simulation time and returns the number of triggered timeouts.
+    size_t advance_time(caf::timespan x) {
+      return clock_.advance_time(x);
+    }
+
+    template <class F>
+    void after_next_enqueue(F f) {
+      after_next_enqueue_ = f;
+    }
+
+    /// Executes the next enqueued job immediately by using the
+    /// `after_next_enqueue` hook.
+    void inline_next_enqueue() {
+      after_next_enqueue([this] { run_once_lifo(); });
+    }
+
+    /// Executes all enqueued jobs immediately by using the `after_next_enqueue`
+    /// hook.
+    void inline_all_enqueues() {
+      after_next_enqueue([this] { inline_all_enqueues_helper(); });
+    }
+
+    bool detaches_utility_actors() const override {
+      return false;
+    }
+
+    test_actor_clock& clock() noexcept override {
+      return clock_;
+    }
+
+  protected:
+    void start() override {
+      dummy_worker worker{this};
+      caf::actor_config cfg{&worker};
+      auto& sys = system();
+      utility_actors_[printer_id] = caf::make_actor<dummy_printer, caf::actor>(
+        sys.next_actor_id(), sys.node(), &sys, cfg);
+    }
+
+    void stop() override {
+      while (run() > 0)
+        trigger_timeouts();
+    }
+
+    void enqueue(caf::resumable* ptr) override {
+      jobs.push_back(ptr);
+      if (after_next_enqueue_ != nullptr) {
+        CAF_LOG_DEBUG("inline this enqueue");
+        std::function<void()> f;
+        f.swap(after_next_enqueue_);
+        f();
```

Comment:
We could spare a few bytes here. 
```suggestion
        after_next_enqueue_();
        after_next_enqueue_ = std::function<void()>{};
```

Reply:
No, we can't do that. The callback might call `after_next_enqueue` again. If we don't swap out before calling, this will result in UB (and a very nasty debugging session).